gui/default/syncthing/core/syncthingController.js

```diff
@@ -3026,6 +3026,55 @@ angular.module('syncthing.core')
+            var failure = $translate.instant("Copy to clipboard failed!");
+
+            if (navigator.clipboard && navigator.clipboard.writeText) {
+                // Default for modern browsers on localhost or HTTPS.
+                navigator.clipboard.writeText(text);
+                return alert(success);
+            } else if (window.clipboardData && window.clipboardData.setData) {
+                // Fallback for Internet Explorer.
+                window.clipboardData.setData('Text', text);
+                return alert(success);
+            } else if (document.queryCommandSupported) {
+                // Fallback for modern browsers on HTTP and non-IE old browsers.
+                // Check for document.queryCommandSupported("copy") support is
+                // omitted on purpose, as old Chrome versions reported "false"
+                // despite supporting the feature.
+                var textarea = document.createElement("textarea");
+                event.currentTarget.appendChild(textarea);
+                textarea.style.position = "fixed";
+                textarea.style.opacity = "0";
+                textarea.textContent = text;
+                textarea.select();
+                try {
+                    document.execCommand("copy");
+                    return alert(success);
+                }
+                catch (ex) {
+                    return alert(failure);
+                }
+                finally {
+                    event.currentTarget.removeChild(textarea);
+                }
+            }
+            return alert(failure);
+        };
+
+        $scope.shareDeviceIdBy = function(channel) {
+            var body = $translate.instant("You have been invited to connect with {%name%} using Syncthing!", {name: $scope.currentDevice.name}) + "\r\n\r\n" + $translate.instant("Accept the invitation by adding a new device using the following ID.") + $translate.instant("If unsure what to do, continue reading below.") + "\r\n\r\n" + $scope.currentDevice.id + "\r\n\r\n" + $translate.instant("Syncthing is a continuous file synchronization program.") + $translate.instant("It synchronizes files between two or more computers in real time, safely protected from prying eyes.") + $translate.instant("Your data is your data alone and you deserve to choose where it is stored, whether it is shared with some third party, and how it's transmitted over the internet.") + "\r\n\r\n" + $translate.instant("For more information about Syncthing please visit https://syncthing.net.");
+
+            if (channel == 'email') {
+                var subject = $translate.instant("Invitation to connect with {%name%} using Syncthing", {name: $scope.currentDevice.name});
+                location.href = "mailto:?subject=" + encodeURIComponent(subject) + "&body=" + encodeURIComponent(body);
+            } else if (channel == 'sms') {
+                location.href = "sms://;?&body=" + encodeURIComponent(body);
```

Comment:
> That's 4 SMS messages already.

I think it will actually be more when writing in non-Latin scripts. The whole body text should probably be reduced to a minimum. I will try to work on this later.

The `&` is required though. For the record, the sms code itself is based on https://stackoverflow.com/a/58131833.

Reply:
I have greatly reduced the SMS text to the required minimum. It takes exactly 146 characters now, but this is with the `{%name%}` placeholder. The actual device name is likely to be longer than that, but we should still have ~10 characters to spare. The main problem is the device ID itself, which normally uses 56 characters, so I've decided to strip the optional hyphens from it.